app/pages/project/workflow-selection.jsx

```diff
@@ -96,6 +89,23 @@ class WorkflowSelection extends React.Component {
         this.setState({ error, loadingSelectedWorkflow: false });
       }
     })
+    .then((workflow) => {
+      if (workflow) {
+        const isWorkflowForProject = workflow.links.project === this.props.project.id;
+        return isWorkflowForProject ? workflow : null;
+      }
+      return null;
+    })
+    .then ((workflow) => {
+      if (workflow) {
+        if (activeFilter) {
+          return workflow.active ? workflow : null;
+        } else {
+          return workflow;
+        }
+      }
+      return null;
+    })
```

Comment:
The project resource's `links.active_workflows` could be used for the activeness check and the project link check before requesting the workflow. We could reduce unnecessary requests this way. I checked Notes from Nature and I see 139 id strings in that serialized property. The `links.workflows` property has 161, so I don't think we're running into max length limits.

Reply:
Good idea. It doesn't make a lot of sense to download the entire workflow, then throw it away.

Reply:
If the project owner is testing an inactive workflow, should we just skip the checks completely?

Reply:
Yes, I think owners, collaborators, expert, tester, and admins should be able to skip activeness check. We'll probably still want to check the workflow belongs to the project though.

Reply:
I've added a check that compares with either `project.links.workflows` or `project.links.active_workflows`, depending on `activeFilter`.